src/JSON/ZeroCopy/Deserializer.dfy

```diff
@@ -188,15 +194,36 @@ module {:options "-functionSyntax:4"} JSON.ZeroCopy.Deserializer {
     {
       var suffixed := Suffixed(elem.t, NonEmpty(sep.t));
       var elems' := SP(elems.t + [suffixed], sep.cs); // DISCUSS: Moving this down doubles the verification time
-      SpecProperties.ConcatBytes_Linear(elems.t, [suffixed], SuffixedElementSpec);
+
+      assert cs0.Bytes() == SuffixedElementsSpec(elems'.t) + sep.cs.Bytes() by {
+        assert {:focus} cs0.Bytes() == SuffixedElementsSpec(elems.t) + (ElementSpec(suffixed.t) + Spec.CommaSuffix(suffixed.suffix)) + sep.cs.Bytes() by {
+          assert cs0.Bytes() == SuffixedElementsSpec(elems.t) + ElementSpec(suffixed.t) + Spec.CommaSuffix(suffixed.suffix) + sep.cs.Bytes() by {
+            assert cs0.Bytes() == SuffixedElementsSpec(elems.t) + elems.cs.Bytes();
+            assert elems.cs.Bytes() == ElementSpec(suffixed.t) + elem.cs.Bytes();
+            assert elem.cs.Bytes() == Spec.CommaSuffix(suffixed.suffix) + sep.cs.Bytes();
+            Seq.Assoc'(SuffixedElementsSpec(elems.t), ElementSpec(suffixed.t), elem.cs.Bytes());
+            Seq.Assoc'(SuffixedElementsSpec(elems.t) + ElementSpec(suffixed.t), Spec.CommaSuffix(suffixed.suffix), sep.cs.Bytes());
+          }
+          Seq.Assoc(SuffixedElementsSpec(elems.t), ElementSpec(suffixed.t), Spec.CommaSuffix(suffixed.suffix));
+        }
+        assert SuffixedElementsSpec(elems.t) + (ElementSpec(suffixed.t) + Spec.CommaSuffix(suffixed.suffix)) + sep.cs.Bytes() == SuffixedElementsSpec(elems'.t) + sep.cs.Bytes() by {
+          assert SuffixedElementsSpec(elems.t) + SuffixedElementSpec(suffixed) == SuffixedElementsSpec(elems.t + [suffixed]) by {
+            SpecProperties.ConcatBytes_Linear(elems.t, [suffixed], SuffixedElementSpec);
+            assert Spec.ConcatBytes(elems.t, SuffixedElementSpec) + Spec.ConcatBytes([suffixed], SuffixedElementSpec) ==  Spec.ConcatBytes(elems.t + [suffixed], SuffixedElementSpec);
+          }
+        }
+      }
+      assert elems'.StrictlySplitFrom?(cs0, SuffixedElementsSpec);
+      assert forall e | e in elems'.t :: e.suffix.NonEmpty? by { assert elems'.t == elems.t + [suffixed]; }
+      assert {:split_here} elems'.cs.Length() < elems.cs.Length();
```

Comment:
I tried to apply the same calc technique there but couldn't arrive to something that would verify quickly enough.